<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1064</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Praktikum 1</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="vbControls_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="vbControls">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinimumSize</enum>
         </property>
         <item>
          <layout class="QVBoxLayout" name="gbRotation">
           <item>
            <layout class="QFormLayout" name="rotations">
             <item row="1" column="0">
              <widget class="QSlider" name="hsRotationA">
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="labelA">
               <property name="text">
                <string>Ring A</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QSlider" name="hsRotationB">
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLabel" name="labelB">
               <property name="text">
                <string>Ring B</string>
               </property>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QSlider" name="hsRotationC">
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QLabel" name="labelC">
               <property name="text">
                <string>Ring C</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="checkBoxes">
           <item>
            <widget class="QCheckBox" name="cbAutoRotate">
             <property name="text">
              <string>Rotate</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="cbCenter">
             <property name="text">
              <string>Center</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="gbProjection">
           <item>
            <widget class="QRadioButton" name="rbPerspective">
             <property name="text">
              <string>Perspective</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rbOrthogonal">
             <property name="text">
              <string>Orthogonal</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="gbClipping">
           <item>
            <layout class="QFormLayout" name="clipping">
             <item row="0" column="1">
              <widget class="QDoubleSpinBox" name="dsbNear">
               <property name="maximum">
                <double>5000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QDoubleSpinBox" name="dsbFar">
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>5000.000000000000000</double>
               </property>
               <property name="value">
                <double>1000.000000000000000</double>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="labelNear">
               <property name="text">
                <string>Near</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="labelFar">
               <property name="text">
                <string>Far</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="sliders">
           <item>
            <layout class="QVBoxLayout" name="slider2">
             <item>
              <widget class="QLabel" name="labelAngle">
               <property name="text">
                <string>Angle</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spAngle">
               <property name="maximum">
                <number>360</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsAngle">
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="slider1">
             <item>
              <widget class="QLabel" name="labelFOV">
               <property name="text">
                <string>FOV</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spFOV">
               <property name="minimum">
                <number>45</number>
               </property>
               <property name="maximum">
                <number>180</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsFOV">
               <property name="minimum">
                <number>45</number>
               </property>
               <property name="maximum">
                <number>180</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QFormLayout" name="flControlButtons"/>
         </item>
         <item>
          <widget class="QPushButton" name="rbReset">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="MyGLWidget" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>480</width>
          <height>360</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>4</width>
          <height>3</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">myglwidget.hpp</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>vsFOV</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>505</y>
    </hint>
    <hint type="destinationlabel">
     <x>172</x>
     <y>452</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spFOV</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vsFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>459</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>507</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vsAngle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>498</y>
    </hint>
    <hint type="destinationlabel">
     <x>52</x>
     <y>449</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spAngle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vsAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>95</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>539</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
